lib/ansible/plugins/action/net_template.py

```diff
@@ -93,6 +93,16 @@ def _handle_template(self):
         except IOError:
             return dict(failed=True, msg='unable to load src file')
 
+        # Create a template search path in the following order:
+        # [working_path, _role_path, *_block._role_paths, dirname(source)]
+        searchpath = [working_path]
+        if self._task._role is not None:
+            searchpath.append(self._task._role._role_path)
+            if self._task._block._dep_chain:
```

Comment:
I think you want to call `get_dep_chain` here, as `_dep_chain` only looks in the block called on, and if there is some recursion going on, `_dep_chain` could be `None` while `get_dep_chain` returns a value. I'm not super clear on the specifics of the code, but that seems like it might be an issue. It is possible I am just confused.